packages/components/src/templates/next/components/complex/ContactInformation/components/ContactMethod.tsx

```diff
@@ -0,0 +1,115 @@
+import { BiEnvelope } from "react-icons/bi"
+
+import type { commonContactMethodStyles } from "./common"
+import type { ContactInformationUIProps } from "~/interfaces"
+import { twMerge } from "~/lib/twMerge"
+import { Link } from "~/templates/next/components/internal/Link"
+import {
+  focusVisibleHighlight,
+  isEmail,
+  isExternalUrl,
+  isPhoneNumber,
+  isUrl,
+  sanitizePhoneNumber,
+} from "~/utils"
+import { METHODS_MAPPING } from "./mapping"
+
+type ContactMethodProps = ContactInformationUIProps["methods"][number] & {
+  styles: ReturnType<typeof commonContactMethodStyles>
+  LinkComponent: ContactInformationUIProps["LinkComponent"]
+}
+
+export const ContactMethod = ({
+  styles,
+  method,
+  label,
+  values,
+  caption,
+  LinkComponent,
+}: ContactMethodProps) => {
+  const methodMapping = method ? METHODS_MAPPING[method] : undefined
+  const Icon = methodMapping?.Icon ?? BiEnvelope
+
+  return (
+    <div className={styles.container()}>
+      <Icon
+        className={
+          methodMapping?.color
+            ? twMerge(styles.icon(), methodMapping.color)
+            : styles.icon()
+        }
+      />
+      <div className={styles.textContainer()}>
+        <div className={styles.label()}>{label}</div>
+        <div className={styles.valuesAndCaptionContainer()}>
+          {values.map((value) => {
+            if (isUrl(value)) {
+              const isExternalLink = isExternalUrl(value)
+              return (
+                <Link
+                  href={value}
+                  isExternal={isExternalLink}
+                  showExternalIcon={isExternalLink}
+                  LinkComponent={LinkComponent}
+                  className={styles.value({
+                    isLink: true,
+                    className: focusVisibleHighlight(),
+                  })}
+                >
+                  {value}
+                </Link>
+              )
+            }
+            if (isEmail(value)) {
+              return (
+                <Link
```

Comment:
issue: I think this and the phone number one should still provide `LinkComponent`, as that is used by Studio to prevent any links inside the preview from being clickable